src/Resources/app/storefront/src/js/filter-slider-range.plugin.js

```diff
@@ -3,334 +3,437 @@ import DomAccess from 'src/helper/dom-access.helper';
+        }
+    }
+
+    refreshDisabledState(filter) {
+        const properties = filter[this.options.name];
+        const entities = properties.entities;
+
+        if (!entities) {
+            this.disableFilter();
+
+            return;
+        }
+
+        const property = entities.find(entity => entity.translated.name === this.options.propertyName);
+        const totalRangePrices = property.options[0].totalRange;
+        const alreadySelectedPrices = this.getValues();
+
+        if (totalRangePrices.min === totalRangePrices.max) {
+            this.disableFilter();
+
+            return;
+        }
+
+        if (this.options.price.min !== totalRangePrices.min || this.options.price.max !== totalRangePrices.max) {
+            this.updateMinAndMaxValues(totalRangePrices.min, totalRangePrices.max);
+        } else {
+            this.enableFilter();
+
+            return;
+        }
+
+        this.updateSelectedRange(alreadySelectedPrices, totalRangePrices);
+
+        this.enableFilter();
+    }
+
+    updateMinAndMaxValues(minPrice, maxPrice) {
+        this.options.price.min = minPrice;
+        this.options.price.max = maxPrice;
+
+        this.slider.noUiSlider.updateOptions({
+            range: {
+                'min': minPrice,
+                'max': maxPrice,
+            },
+        });
+
+        this.updateInputsAndSliderValues(minPrice, maxPrice);
+    }
+
+    updateInputsAndSliderValues(minPrice, maxPrice) {
+        if (minPrice !== null) {
+            this._inputMin.value = minPrice;
+        }
+
+        if (maxPrice !== null) {
+            this._inputMax.value = maxPrice;
+        }
+
+        this.setBothKnobValues();
+    }
+
+    /**
+     * @param {Array} alreadySelectedPrices
+     * @param {Object} totalRangePrices
+     */
+    updateSelectedRange(alreadySelectedPrices, totalRangePrices) {
+        const alreadySelectedMin = alreadySelectedPrices[this.options.minKey];
+        const alreadySelectedMax = alreadySelectedPrices[this.options.maxKey];
+
+        if (alreadySelectedMin >= totalRangePrices.min && alreadySelectedMax <= totalRangePrices.max) {
+            this.updateInputsAndSliderValues(alreadySelectedMin, alreadySelectedMax);
+        } else if (alreadySelectedMin < totalRangePrices.min && alreadySelectedPrices[this.options.maxKey] <= totalRangePrices.max) {
+            this.updateInputsAndSliderValues(null, alreadySelectedMax);
+        } else if (alreadySelectedMin >= totalRangePrices.min && alreadySelectedMax > totalRangePrices.max) {
+            this.updateInputsAndSliderValues(alreadySelectedMin, null);
+        } else if (typeof alreadySelectedMin === 'undefined' && alreadySelectedMax <= totalRangePrices.max) {
+            this.updateInputsAndSliderValues(null, alreadySelectedMax);
+        } else if (alreadySelectedMin >= totalRangePrices.min && typeof alreadySelectedMax === 'undefined') {
+            this.updateInputsAndSliderValues(alreadySelectedMin, null);
+        }
```

Comment:
Something like this should do the same and look a bit clearer:

```suggestion
        const updateMin = alreadySelectedMin && alreadySelectedMin >= totalRangePrices.min;
        const updateMax = alreadySelectedMax && alreadySelectedMax <= totalRangePrices.max;

        const newSelectedMin = updateMin ? alreadySelectedMin : null;
        const newSelectedMax = updateMax ? alreadySelectedMax : null;

        this.updateInputsAndSliderValues(newSelectedMin, newSelectedMax);
```